scripts/figma-import/icons.js

```diff
@@ -0,0 +1,78 @@
+const config = require('../../figma-import.config')
+
+const path = require('path')
+const fs = require('fs').promises
+const { paramCase } = require('change-case')
+
+const { getImageContent, getNodeChildren, getSvgImageUrl } = require('./utils/api')
+
+const iconsDir = path.resolve('src/shared/icons/figmaSvgs')
+
+let counter = 0
+
+/**
+ * clear icons dir
+ *
+ */
+const clearIconsDir = async () => {
+  try {
+    await fs.rm(iconsDir, { recursive: true })
+    console.log(`${iconsDir} successfully deleted!`)
+  } catch (err) {
+    console.error(`Error while deleting ${iconsDir}`)
+  }
+}
+
+/**
+ * generate icon content
+ * [iconName].svg files
+ *
+ * @param {object} iconNode
+ * @param {number} total
+ * @return {Promise<void>}
+ */
+const generateIcon = async (iconNode, total) => {
+  const iconUrl = await getSvgImageUrl(iconNode.id)
+
+  const iconName = paramCase(iconNode.name)
+
+  try {
+    await fs.access(iconsDir)
+  } catch (error) {
+    await fs.mkdir(iconsDir)
+  }
+
+  const { data: iconContent } = await getImageContent(iconUrl)
+
+  await Promise.all([await fs.writeFile(path.resolve(iconsDir, `${iconName}.svg`), iconContent, 'utf-8')])
+
+  counter++
+  process.stdout.write(` ${counter}/${total} icons has been saved\r`)
+  if (counter === total) {
+    console.info('All svgs has been saved')
+  }
+}
+
+/**
+ * generate icons components
+ *
+ * @param {[Object]} iconNodesArr - array of icon nodes from frame
+ * @return {Promise<void>}
+ */
+const generateIcons = async (iconNodesArr) => {
+  await Promise.all(
+    iconNodesArr.map((item, index) => {
+      setTimeout(() => generateIcon(item, iconNodesArr.length), 1000 * index)
```

Comment:
What's happening here? Seems we're forcing this process to take at least 3 minutes (we have ~180 icons)

Reply:
We don't have choice due to figma api limitations. Without this or with smaller timeout value this process is failing.

Reply:
Ah, so this is the solution to the too many requests problem you've mentioned?

Reply:
Exactly

Reply:
All right, sounds reasonable